test/test_common/utility.h

```diff
@@ -461,10 +461,6 @@ class TestUtility {
 
   static constexpr std::chrono::milliseconds DefaultTimeout = std::chrono::milliseconds(10000);
 
-  static void renameFile(const std::string& old_name, const std::string& new_name);
-  static void createDirectory(const std::string& name);
-  static void createSymlink(const std::string& target, const std::string& link);
```

Comment:
just to make sure I understand, was the movement from here to environment just to get all the filesystem stuff in the same file for clarity? Or was there some other reason?

Reply:
Yes, and to deduplicate as `TestEnvironment` also has a `createPath` which is equivalent to `createDirectory`

Reply:
> just to make sure I understand, was the movement from here to environment just to get all the filesystem stuff in the same file for clarity? Or was there some other reason?

Yes, and to deduplicate as `TestEnvironment` also has a `createPath` which is equivalent to `createDirectory`

Reply:
Additionally, consuming functions between TestUtility and TestEnvironment introduced a cyclic dependency graph between TestEnvironment, TestNetworkUtility, TestUtility (and one other I forget offhand.) TestEnvironment must consume TestUtility, so the few exceptions that were cyclic in TestUtility were moved to TestEnvironment rather than maintain duplicate functionality.